File: typescript-sdk/src/sdk/models/shared/room.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";


export enum RoomTypeEnum {
    Room = "room"
}


export class Room extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=createdAt" })
  createdAt?: Date;

  @SpeakeasyMetadata({ data: "json, name=defaultAccesses" })
  defaultAccesses?: any;

  @SpeakeasyMetadata({ data: "json, name=groupsAccesses" })
  groupsAccesses?: Record<string, any>;

  @SpeakeasyMetadata({ data: "json, name=id" })
  id?: string;

  @SpeakeasyMetadata({ data: "json, name=lastConnectionAt" })
  lastConnectionAt?: Date;

  @SpeakeasyMetadata({ data: "json, name=metadata" })
  metadata?: Record<string, string>;

  @SpeakeasyMetadata({ data: "json, name=type" })
  type?: RoomTypeEnum;

  @SpeakeasyMetadata({ data: "json, name=usersAccesses" })
  usersAccesses?: Record<string, any>;
}

File: go-sdk/pkg/models/shared/room.go
package shared

import (
	"time"
)

type RoomTypeEnum string

const (
	RoomTypeEnumRoom RoomTypeEnum = "room"
)

type Room struct {
	CreatedAt        *time.Time             `json:"createdAt,omitempty"`
	DefaultAccesses  *interface{}           `json:"defaultAccesses,omitempty"`
	GroupsAccesses   map[string]interface{} `json:"groupsAccesses,omitempty"`
	ID               *string                `json:"id,omitempty"`
	LastConnectionAt *time.Time             `json:"lastConnectionAt,omitempty"`
	Metadata         map[string]string      `json:"metadata,omitempty"`
	Type             *RoomTypeEnum          `json:"type,omitempty"`
	UsersAccesses    map[string]interface{} `json:"usersAccesses,omitempty"`
}
